pod/perlre.pod

```diff
@@ -874,41 +874,88 @@ inconsistencies (bugs) with the C</d> modifier, where Unicode rules
 would be used when inappropriate, and vice versa.  C<\p{}> did not imply
 Unicode rules, and neither did all occurrences of C<\N{}>, until 5.12.
 
-=head2 Regular Expressions
+=head2 Regexen, Regexps and Regular Expressions
```

Comment:
I don't particularly agree with the relevance of this section here (in addition to my previous objections to renaming sections unnecessarily)

Reply:
To try to be more specific: this seems like a subject for a blog post, rather than Perl documentation on regexes.

Reply:
(at the very least, I would prefer this be a separate task from this PR's useful reorganization)